File: src/test/java/de/schramm/royalbash/controller/service/core/card/CardMock.java
package de.schramm.royalbash.controller.service.core.card;

import de.schramm.royalbash.controller.service.core.Card;
import de.schramm.royalbash.controller.service.core.Context;
import de.schramm.royalbash.controller.service.core.Game;
import lombok.Builder;
import lombok.Value;

@Value
@Builder
public class CardMock implements Card {

    private final String id;
    private final String name;
    private final int cost;

    private final boolean placeableOnSpot = false;

    @Override
    public Game invoke(final Context context) {
        return context.getGame();
    }
}

File: src/test/kotlin/de/schramm/royalbash/controller/service/core/card/CardMock.kt
package de.schramm.royalbash.controller.service.core.card

import de.schramm.royalbash.controller.service.core.Card
import de.schramm.royalbash.controller.service.core.Context
import de.schramm.royalbash.controller.service.core.Game

class CardMock(
        override val id: String,
        override val name: String = "Card Mock",
        override val text: String = "Card Mock Text",
        override val cost: Int = 0
) : Card {

    override fun invoke(context: Context): Game {
        return context.game
    }
}
